internal/tools/pipeline/actionagent/step_prepare_test.go

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package actionagent
+
+import (
+	"os"
+	"testing"
+
+	"bou.ke/monkey"
+
+	"github.com/erda-project/erda/pkg/filehelper"
+)
+
+func Test_setupScript(t *testing.T) {
+	tests := []struct {
+		name     string
+		commands interface{}
+		got      string
+		want     string
+	}{
+		{
+			name:     "string slice commands",
+			commands: []string{"sleep 120"},
+			want: `#!/bin/sh
+set -e
+
+echo + "sleep 120"
+sleep 120 || ((echo "- FAIL! exit code: $?") && false)
+echo
+
+`,
+		},
+		{
+			name: "string commands",
+			commands: `npm i @terminus/trnw-cli -g
+            trnw-cli build apk -T -A -t assembleRelease \
+                        --proxy \
+                        --env-default mall-((PROJECT_TYPE))/.env.default`,
+			want: `#!/bin/sh
+set -e
+
+echo + "npm i @terminus/trnw-cli -g\n            trnw-cli build apk -T -A -t assembleRelease \\\n                        --proxy \\\n                        --env-default mall-((PROJECT_TYPE))/.env.default"
+npm i @terminus/trnw-cli -g
+            trnw-cli build apk -T -A -t assembleRelease \
+                        --proxy \
+                        --env-default mall-((PROJECT_TYPE))/.env.default || ((echo "- FAIL! exit code: $?") && false)
```

Comment:
I think block script should be different with multi-line commands.
We should let the use know which concrete line in block script fails.

Reply:
if the user wrote block script, it's difficult to split each command
for instance:
```
git clone xxx \
-depth=1 \
-v
```

```
cat /tmp/xxx
sleep 10
```